### PR TITLE
Support for more examples from the 'peripheral' category

### PR DESCRIPTION
**Pull request by borys-jelenski-polidea**
Added examples from the 'peripheral' category: bsp, fatfs, flash_fds, flash_fstorage, flashwrite, gpiote, i2s, led_softblink, low_power_pwm, lpcomp, pin_change_int, ppi, pwm_driver, qdec, qspi, rng, rtc, saadc, serial, simple_timer, temperature, template_project, timer, usbd_hid_composite, usbd_hid_generic, usbd_msc, wdt

Added libs: nrf5_app_pwm, nrf5_app_simple_timer, nrf5_app_sdcard, nrf5_ext_protothreads, nrf5_ext_fatfs, nrf5_ext_fatfs_port_diskio_blkdev, nrf5_block_dev_empty, nrf5_block_dev_qspi, nrf5_block_dev_ram, nrf5_block_dev_sdc, nrf5_low_power_pwm, nrf5_led_softblink, nrf5_serial, nrf5_crc32, nrf5_nrfx_ppi, nrf5_drv_ppi, nrf5_drv_gpiote, nrf5_nrfx_timer, nrf5_drv_timer, nrf5_nrfx_qspi, nrf5_drv_qspi, nrf5_nrfx_i2s, nrf5_drv_i2s, nrf5_nrfx_comp, nrf5_drv_comp, nrf5_nrfx_lpcomp, nrf5_drv_lpcomp, nrf5_nrfx_saadc, nrf5_drv_saadc, nrf5_nrfx_pwm, nrf5_drv_pwm, nrf5_nrfx_wdt, nrf5_drv_wdt, nrf5_nrfx_qdec, nrf5_drv_qdec

### CODE REVIEW

File: ci/libraries/nrf5_external.json [hunk 1]
```diff
@@ -324,5 +324,47 @@
...
+        "external/protothreads",
+        "external/protothreads/pt-1.4"
+      ]
+    }
+  },
+  "nrf5_ext_fatfs": {
+    "documentation": "FatFs - Generic FAT file system module",
+    "variant": "object",
+    "sources": [
+      "/external/fatfs/src/ff.c"
```
Comment (Cierpliwy) [inline ci/libraries/nrf5_external.json:342]: Small nit: extra `/` in the path.

File: ci/libraries/nrf5_external.json [hunk 2]
```diff
@@ -324,5 +324,47 @@
...
+      "public": [
+        "/external/fatfs/src"
+      ]
+    }
+  },
+  "nrf5_ext_fatfs_port_diskio_blkdev": {
+    "documentation": "FatFS disk I/O interface based on block device",
+    "variant": "object",
+    "sources": [
+      "/external/fatfs/port/diskio_blkdev.c"
```
Comment (Cierpliwy) [inline ci/libraries/nrf5_external.json:354]: Small nit: extra `/` in the path.

File: ci/libraries/nrf5_external.json [hunk 3]
```diff
@@ -324,5 +324,47 @@
...
+      "public": [
+        "nrf5_ext_fatfs",
+        "nrf5_block_dev",
+        "nrf5_mdk",
+        "nrf5_soc"
+      ]
+    },
+    "includes": {
+      "public": [
+        "/external/fatfs/port"
```
Comment (Cierpliwy) [inline ci/libraries/nrf5_external.json:366]: Small nit: extra `/` in the path.

File: ci/libraries/nrf5_misc.json [hunk 4]
```diff
@@ -108,13 +108,204 @@
       ]
     }
   },
+  "nrf5_app_sdcard": {
```
Comment (Cierpliwy) [inline ci/libraries/nrf5_misc.json:111]: Not in nrf5_app.json?

File: ci/libraries_tests/nrf5_app_sdcard/CMakeLists.txt [hunk 5]
```diff
@@ -19,85 +19,50 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
+#
+# WARNING: FILE GENERATED FROM ./ci/scripts/generate_cmake_tests.sh SCRIPT.
```
Comment (Cierpliwy) [inline ci/libraries_tests/nrf5_app_sdcard/CMakeLists.txt:23]: Hmm? What happened here?

File: ci/libraries_tests/nrf5_app_usbd_hid_generic/main.c [hunk 6]
```diff
@@ -1,12 +0,0 @@
-#include "app_usbd_hid_generic.h"
```
Comment (Cierpliwy) [inline ci/libraries_tests/nrf5_app_usbd_hid_generic/main.c:1]: Why removed?

File: ci/libraries_tests/nrf5_serial/CMakeLists.txt [hunk 7]
```diff
@@ -23,7 +23,7 @@
 # WARNING: FILE GENERATED FROM ./ci/scripts/generate_cmake_tests.sh SCRIPT.
 
 cmake_minimum_required(VERSION 3.14)
-project(nrf5_app_usbd_hid_generic_test LANGUAGES C ASM)
+project(nrf5_serial_test LANGUAGES C ASM)
```
Comment (Cierpliwy) [inline ci/libraries_tests/nrf5_serial/CMakeLists.txt:26]: Once again, I'm not sure why move was detected..

File: ci/scripts/build_all_examples.sh [hunk 8]
```diff
@@ -75,7 +75,7 @@ function build_all_configs() {
             if [[ $board_sdk_ignore_config_list =~ "${board}_${sdk_version}" ]]; then
                 continue
             fi
-            if [[ $board_sd_ignore_config_list =~ "${board}_${real_sd_variant}" ]]; then
+            if [[ $board_sd_ignore_config_list =~ "${board}_${sd_variant}" ]]; then
```
Comment (Cierpliwy) [inline ci/scripts/build_all_examples.sh:78]: Good catch!

File: ci/libraries_tests/nrf5_app_usbd_hid_generic/main.c (hunk 6 above)
Comment (borys-jelenski-polidea) [reply ci/libraries_tests/nrf5_app_usbd_hid_generic/main.c:1]: Actually, I cannot build this example due to a snipped of code you posted on Slack:
```C
CRITICAL_REGION_ENTER();
ret_code_t ret = app_usbd_ep_transfer(ep_addr, &transfer);
if (ret == NRF_SUCCESS)
{
    app_usbd_hid_state_flag_set(&p_generic_ctx->hid_ctx,
                                APP_USBD_HID_STATE_FLAG_TRANS_IN_PROGRESS);
}
CRITICAL_REGION_EXIT();

return ret;
```
The problem is, the `CRITICAL_REGION_ENTER/EXIT()` pair create an additional scope but only when we are building with `SOFTDEVICE_PRESENT`, which we do when building tests for libs. How did you fix that?

File: ci/libraries_tests/nrf5_app_sdcard/CMakeLists.txt (hunk 5 above)
Comment (borys-jelenski-polidea) [reply ci/libraries_tests/nrf5_app_sdcard/CMakeLists.txt:23]: What do you mean? I assumed you add tests for libraries by running generate_cmake_script.sh, tweaking the generated CMakeLists.txt and adding main.c and sdk_config.h.

File: ci/libraries/nrf5_misc.json (hunk 4 above)
Comment (borys-jelenski-polidea) [reply ci/libraries/nrf5_misc.json:111]: Yeah, probably it should be there.

File: ci/libraries_tests/nrf5_app_sdcard/CMakeLists.txt (hunk 5 above)
Comment (Cierpliwy) [reply ci/libraries_tests/nrf5_app_sdcard/CMakeLists.txt:23]: I was confused about move operation. I saw that you removed one example and that's the reason.

File: ci/libraries_tests/nrf5_app_usbd_hid_generic/main.c (hunk 6 above)
Comment (Cierpliwy) [reply ci/libraries_tests/nrf5_app_usbd_hid_generic/main.c:1]: Prepare script contains a patch for the SDK (look at ./ci/patches folder).

**Pull request merged**